devito/operator.py

```diff
@@ -295,7 +297,13 @@ def _schedule_expressions(self, clusters, ordering):
                 needed = entries[index:]
 
                 # Build and insert the required Iterations
-                iters = [Iteration([], j.dim, j.dim.size, offsets=j.ofs) for j in needed]
+                iters = []
+                for j in needed:
```

Comment:
I think this should just work with a single list comprehension as before, using ``symbolic_size``, and w/o using any ``if ... else``. If it doesn't, can you point me to one failing example?

Reply:
This is changed again in the next PR. Goes back to being a one-liner but written in terms of limits. Can you wait for PR 2/3 before you come to a conclusion on what's the best way?

Reply:
Unlike the previous ones, this bit of is of fundamental importance, so could you shed more light on this?

Reply:
Fixed